clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -3815,21 +3815,23 @@ private void recordNearlineLocalBrokerToReadyToServerLatency(
       KafkaMessageEnvelope kafkaMessageEnvelope,
       LeaderProducedRecordContext leaderProducedRecordContext) {
     /**
-     * Record nearline latency only when it's a hybrid store and the lag has been caught up. Sometimes
-     * the producerTimestamp can be -1 if the leaderProducedRecordContext had an error after callback
-     * Don't record latency for invalid timestamps
+     * Record nearline latency only when it's a hybrid store, the lag has been caught up and ignore
+     * messages that are getting caughtup. Sometimes the producerTimestamp can be -1 if the
+     * leaderProducedRecordContext had an error after callback. Don't record latency for invalid timestamps.
      */
     if (!isUserSystemStore() && isHybridMode() && partitionConsumptionState.hasLagCaughtUp()) {
-      long afterProcessingRecordTimestampMs = System.currentTimeMillis();
       long producerTimestamp = (leaderProducedRecordContext == null)
           ? kafkaMessageEnvelope.producerMetadata.messageTimestamp
           : leaderProducedRecordContext.getProducedTimestampMs();
       if (producerTimestamp > 0) {
```

Comment:
Nitpick: I think you could replace this check entirely with `partitionConsumptionState.isNearlineMetricsRecordingValid` since that function includes `producerTimeStampInMs > lagCaughtUpTimeInMs` which is equivalent... so no need for two nested ifs.

Reply:
Else part of this logs the invalid timestamps. Didn't want to log the else case for `isNearlineMetricsRecordingValid`, so added a nesting here. What do you think?

Reply:
Oh yeah, sure, I missed that.